atomic_reactor/plugins/post_push_to_pulp.py

```diff
@@ -109,47 +152,32 @@ def push_tar(self, filename, image_names=None, repo_prefix="redhat-"):
 
         pulp_repos = self.pulp_handler.create_dockpulp_and_repos(image_names, repo_prefix)
         _, file_extension = os.path.splitext(filename)
+        upload_filename = None
 
         try:
             top_layer, layers = self.pulp_handler.get_tar_metadata(filename)
-            # getImageIdsExist was introduced in rh-dockpulp 0.6+
-            existing_imageids = self.pulp_handler.get_image_ids_existing(layers)
-            self.log.debug("existing layers: %s", existing_imageids)
-
-            # Strip existing layers from the tar and repack it
-            remove_layers = [str(os.path.join(x, 'layer.tar')) for x in existing_imageids]
-
-            commands = {'.xz': 'xzcat', '.gz': 'zcat', '.bz2': 'bzcat', '.tar': 'cat'}
-            unpacker = commands.get(file_extension, None)
-            self.log.debug("using unpacker %s for extension %s", unpacker, file_extension)
-            if unpacker is None:
-                raise Exception("Unknown tarball format: %s" % filename)
-
-            with NamedTemporaryFile(prefix='strip_tar_', suffix='.gz') as outfile:
-                cmd = "set -o pipefail; {0} {1} | tar --delete {2} | gzip - > {3}".format(
-                    unpacker, filename, ' '.join(remove_layers), outfile.name)
-                self.log.debug("running %s", cmd)
-                subprocess.check_call(cmd, shell=True)
-                self.log.debug("uploading %s", outfile.name)
-                self.pulp_handler.upload(outfile.name)
+            upload_filename = self._deduplicate_layers(layers, filename, file_extension)
         except:
             self.log.debug("Error on creating deduplicated layers tar", exc_info=True)
             try:
                 if file_extension != '.tar':
                     raise RuntimeError("tar is already compressed")
-                with NamedTemporaryFile(prefix='full_tar_', suffix='.gz') as outfile:
-                    cmd = "set -o pipefail; cat {0} | gzip - > {1}".format(
-                        filename, outfile.name)
-                    self.log.debug("Compressing tar using '%s' command", cmd)
-                    subprocess.check_call(cmd, shell=True)
-                    self.log.debug("uploading %s", outfile.name)
-                    self.pulp_handler.upload(outfile.name)
+                upload_filename = self._gzip_file(filename)
             except:
                 self.log.info("Falling back to full tar upload")
-                self.pulp_handler.upload(filename)
+
+        in_rh_everything = False
+        if upload_filename:
+            filename = upload_filename
+        for repo_id, pulp_repo in pulp_repos.items():
```

Comment:
This can be just `for repo_id in pulp_repos` since `pulp_repo` is not used in this loop.

Reply:
yop